### ISSUE TITLE
Optima

### ISSUE THREAD

**Issue by guicho271828** [quicklisp/quicklisp-projects#1889]
So I heard if Trivia provides Optima compatibility asd, you can replace Optima.
The pull requesting that achieves it is pending, https://github.com/guicho271828/trivia/pull/115
it was tested on the dependent libraries and there was only minor issues.
https://github.com/guicho271828/trivia/issues/31#issuecomment-667576750
https://github.com/guicho271828/trivia/issues/31#issuecomment-667579962
https://github.com/flambard/cl-erlang-term/pull/5#event-3639461111
thoughts?

**Comment by guicho271828** [comment]
only minor issues -> cl-erlang-term merged in the changes, so there is no issue now

**Comment by phoe** [comment]
What do you mean by "replace Optima"? Do you suggest to remove Optima from Quicklisp altogether and replace it with Trivia's compatibility system?

**Comment by guicho271828** [comment]
yes, see discussion in the issue thread above -- @sionescu suggested it first because of the dependency to the broken `eos` library issue

**Comment by sionescu** [comment]
@phoe yes because Optima is no longer maintained and bug reports won't get fixed.

**Comment by quicklisp** [comment]
There is no reason to think Optima will not get fixed. At the very least it can be adopted by sharplispers and minimally maintained.

So far this "transition" has broken more things that it has fixed.

**Comment by Hexstream** [comment]
> There is no reason to think Optima will not get fixed.

A surprising statement, given that [the Optima repository](https://github.com/m2ym/optima) that [quicklisp pulls from](https://github.com/quicklisp/quicklisp-projects/blob/master/projects/optima/source.txt) has been archived by [its author and primary contributor](https://github.com/m2ym) @m2ym alongside all his other Common Lisp repositories, that the repository has [seen its last commit more than 5 years ago](https://github.com/m2ym/optima/commits/master) and that [Trivia](https://github.com/guicho271828/trivia) has been widely known as Optima's replacement for ages. How about some [consolidation](https://fare.livejournal.com/169346.html)?

> At the very least it can be adopted by sharplispers and minimally maintained.

My impression has always been that [sharplispers](https://github.com/sharplispers) is mostly intended as a last resort if nobody else is interested in maintaining or superseding a repository. Was I mistaken? Wouldn't it be the first time in history that sharplispers adopts a repository for which a clear transition path is actively being built?

**Comment by quicklisp** [comment]
From my perspective:

1. Optima built fine
2. Software that depends on optima continues to build
3. Making changes has broken builds

My order of preference is:

1. Don't make any changes
2. Make minimal changes that preserve building (via sharplispers or similar)
3. Make transitional changes that preserve building (via some compat thing)
4. Make transitional changes that break stuff (the current plan)

The current plan seems to be 4 and I'm not sure the situation is improving over time.

**Comment by phoe** [comment]
Has there been an attempt to build (and preferably test) the Quicklisp world with Optima phased out in favor of Trivia compatibility layer? If yes, how big is the fallout?

**Comment by guicho271828** [comment]
I haven't added `optima.asd` file to the trivia repository yet. If you remove Optima now, then all libraries that depend on optima should fail.
Also: There are a couple of minor wrapper macros in Optima that was missing in Trivia, which was added recently (~july). Depending on the timing you tried (perhaps by manually adding optima.asd) before, you may try it again. btw, those macros are largely undocumented in Optima.

**Comment by guicho271828** [comment]
(it is still in the PR)
